src/components/FavoritesBtn.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState } from 'react';
+
+import { FavoriteMovie } from '../models/index';
+import api from '../service/api';
+
+import '../styles/FavoritesBtn.scss';
+
+const FavoritesBtn: React.FC<FavoriteMovie> = ({ movieId }) => {
+  const [icon, setIcon] = useState(api.isIdInFavorites(movieId));
+
+  const heartUnfilledIcon = '/images/favoriteBtn/heartFilled.svg';
+  const heartFilledIcon = '/images/favoriteBtn/heartUnfilled.svg';
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>): void => {
+    event.preventDefault();
+    api.isIdInFavorites(movieId)
+      ? api.setFavoritesId(movieId)
+      : api.deleteFavoritsId(movieId);
+    setIcon(api.isIdInFavorites(movieId));
+  };
+
+  return (
+    <a href="#!" className="favorite-btn" onClick={handleClick}>
+      <img
+        className="favorite-icon"
+        src={icon ? heartFilledIcon : heartUnfilledIcon}
```

Comment:
is icon boolean type? if yes, choose more appropriate name

Reply:
I agree that the name isn't appropriate. Here, in a state, I would like to store a boolean value that allows to check local storage and based on that data to change the state of a favorite icon. I am thinking of names like favoriteIconState, setFavoriteIconState. is it an appropriate name?

Reply:
maybe you can use name like - "iconStatus" or something similar

Reply:
Good one, but I still believe that I need to stay consistent in terms of  working with favorites icon, hence I have a feeling that favoriteIconState could be more suitable here

Reply:
as a rule boolean variable should start with is prefix, here the most appropriate name from my point of view is isFavorite

Reply:
Thanks for a hint. I have already renamed it.